src/main_menu.cpp

```diff
@@ -243,6 +242,77 @@ std::vector<std::string> main_menu::load_file( const std::string &path,
     return result;
 }
 
+/* Calculate date of Easter - a direct implementation of the "Anonymous Gregorian" aka Meeus/Jones/Butcher Gregorian algorithm, given here:
+    https://en.wikipedia.org/wiki/Computus#Anonymous_Gregorian_algorithm */
+bool main_menu::is_easter( int day, int month, int year )
+{
+    const int a = year % 19;
+    const int b = year / 100;
+    const int c = year % 100;
+    const int d = b / 4;
+    const int e = b % 4;
+    const int f = ( b + 8 ) / 25;
+    const int g = ( b - f + 1 ) / 3;
+    const int h = ( 19 * a + b - d - g + 15 ) % 30;
+    const int i = c / 4;
+    const int k = c % 4;
+    const int l = ( 32 + 2 * e + 2 * i - h - k ) % 7;
+    const int m = ( a + 11 * h + 22 * l ) / 451;
+    const int e_month = ( h + l - 7 * m + 114 ) / 31;
+    const int e_day = ( ( h + l - 7 * m + 114 ) % 31 ) + 1;
+
+    return ( ( month == e_month ) && ( day == e_day ) );
+}
+
+holiday main_menu::get_holiday_from_time()
+{
+    bool success = false;
+
+    std::tm local_time;
+    std::time_t current_time = std::time( nullptr );
+
+    /* necessary to pass LGTM, as threadsafe version of localtime differs by platform */
+#if defined(_WIN32)
+
+    errno_t err = localtime_s( &local_time, &current_time );
+    if( err == 0 ) {
+        success = true;
+    }
+
+#else
+
+    success = !!localtime_r( &current_time, &local_time );
+
+#endif
+
+    if( success ) {
+
+        const int month = local_time.tm_mon + 1;
+        const int day = local_time.tm_mday;
+        const int wday = local_time.tm_wday;
+        const int year = local_time.tm_year + 1900;
+
+        /* check date against holidays */
+        if( month == 1 && day == 1 ) {
+            return holiday::new_year;
+        } else if( month == 3 ||
+                   month == 4 ) { // only run easter date calculation if currently March or April
```

Comment:
Could you move this comment onto the line above or below? It's making this condition go across two lines and look really ugly.